test/controller/download.attachment.landing.controller.spec.unit.ts

```diff
@@ -0,0 +1,69 @@
+jest.mock("ioredis");
+jest.mock("../../src/middleware/authentication.middleware");
+jest.mock("../../src/middleware/session.middleware");
+jest.mock("../../src/services/objection.session.service");
+jest.mock("../../src/middleware/objection.session.middleware");
+
+import { Session } from "ch-node-session-handler/lib/session/model/Session";
+import { NextFunction, Request, Response } from "express";
+import request from "supertest";
+import app from "../../src/app";
+import { OBJECTIONS_SESSION_NAME } from "../../src/constants";
+import authenticationMiddleware from "../../src/middleware/authentication.middleware";
+import objectionSessionMiddleware from "../../src/middleware/objection.session.middleware";
+import sessionMiddleware from "../../src/middleware/session.middleware";
+import { COOKIE_NAME } from "../../src/utils/properties";
+
+const mockAuthenticationMiddleware = authenticationMiddleware as jest.Mock;
+mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next());
+
+const mockSessionMiddleware = sessionMiddleware as jest.Mock;
+mockSessionMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => {
+  req.session = {
+    data: {},
+  } as Session;
+  return next();
+});
+
+const mockObjectionSessionMiddleware = objectionSessionMiddleware as jest.Mock;
```

Comment:
As this will only be accessed by internal staff (I believe) we should probably exclude this middleware from running against this endpoint as an ObjectionSession isn't really useful for them

Reply:
we will need the user's access token from the session (not chSession though? not sure) to call the api. I think this is how we get the eric user info to authorise the download on the api side.

Reply:
looking at the code, we have 2 middleware functions for the session, the objection.session one creates the 'objection session' add on data and won't affect the access token. So you could do as Ian suggested and add an 'if' into the objection.session middleware to return next if it is a download call. You could do it in this PR or a separate one, up to you.

Reply:
if you do manage to bypass the objection.session middleware, you could actually keep that mock and test that the mock isn't called for a download.

Reply:
I think if you change app.ts
`app.use(`${pageURLs.STRIKE_OFF_OBJECTIONS}/*`, objectionSessionMiddleware);` to 
`app.use(`${pageURLs.STRIKE_OFF_OBJECTIONS}/*(?<!download)$`, objectionSessionMiddleware);`
then objectionSessionMiddleware will not run on endpoints ending in download
regexr.com/5a2k8  (If node supports negative lookbehind that is, a quick google suggests it might)